### ISSUE TITLE
Acova Taffetas hvac modes

### ISSUE THREAD

**Issue by molusk** [Koenkk/zigbee-herdsman-converters#4751]
Hi,

I have an issue with Acova Taffetas2 hvac modes.

The configuration is a copy/paste from other Acova heaters so I don't really know what I did and how Taffetas2 works compared to other models but when I set my climate entity in home assistant to "heat" it sets the heater to away (7ºC). 

"Auto" mode enables heating mode with presence detection which is ok for me. But I have no way to set "eco" mode which disables the presence detection sensor. Eco mode is very usefull ton maintain temperature in a bedroom by night, without it the set point decreases by 1º every 2h and I don't think there is a lower limit (which is the eco set point when the heater is not connected). 

My kids bedroom setpoint is arount 15ºC in the morning unless I tell hass to update it regulary overnight.

Finally, the local temperature reported by the heater is in fact the current setpoint. The heater has no temperature reporting function.

To sum up, I am not completely satisfied with this heater intergration and would like to adapt it but I need some pointers to edit the converter.

**Comment by molusk** [comment]
I didn't find a list of available hvac modes in HA/Z2M and don't understand how "heat" mode is mapped to the heater's frost-protection mode.

Regarding the auto mode, I have found a way to expose unoccupied_heating_setpoint and will test if I am able to fix the lower limit with that option. The parameter is only available through Z2M UI which shoud not be an issue as I don't need to play with it (as I have direct access to frost-protection mode).

I tried to expose occupancy reported by the heater to confirm setpoint is impacted by this function but it did not work so far.

In case it could help somebody, here is my custom converter.

```js
// const exposes = require('../lib/exposes');
// const fz = require('../converters/fromZigbee');
// const tz = require('../converters/toZigbee');
// const reporting = require('../lib/reporting');

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const e = exposes.presets;

module.exports = [
	{
//        zigbeeModel: ['TAFFETAS2 D1.00P1.02Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000', 'TAFFETAS2 D1.00P1.01Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000'],
        zigbeeModel: ['TAFFETAS2 D1.00P1.01Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000'],
        model: 'TAFFETAS2',
        vendor: 'Acova',
        description: 'Taffetas 2 heater (custom))',
        fromZigbee: [fz.thermostat, fz.hvac_user_interface],
        toZigbee: [
            tz.thermostat_local_temperature,
            tz.thermostat_system_mode,
            tz.thermostat_occupied_heating_setpoint,
            tz.thermostat_unoccupied_heating_setpoint,
            tz.thermostat_occupied_cooling_setpoint,
            tz.thermostat_running_state,
        ],
        exposes: [
            exposes.climate()
                .withSetpoint('occupied_heating_setpoint', 7, 28, 0.5)
                .withSetpoint('unoccupied_heating_setpoint', 7, 28, 0.5)
                .withLocalTemperature()
                .withSystemMode(['off', 'heat', 'auto'])
                .withRunningState(['idle', 'heat']),
            e.occupancy(),
        ],
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(1);
            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 'hvacThermostat']);
            await reporting.thermostatTemperature(endpoint);
            await reporting.thermostatRunningState(endpoint);
            await reporting.thermostatOccupiedHeatingSetpoint(endpoint);
            await reporting.thermostatUnoccupiedHeatingSetpoint(endpoint);
        },
    },
];
```

**Comment by molusk** [comment]
Small update, I was able to expose the heater’s occupancy sensor but I am still struggling with the open window sensor which seem to be a simple binaryInput. I don’t know how to expose it and can’t find examples yet.

Last but not least, I don’t understand why ha/z2m "heat" mode enables heatear’s frost-protection mode. Or what parameter makes the difference between frost-protection, night and day modes other than "local_temperature".
Is it possible to just change the mode name for this heater ? (like fzlocal functions for some devices)

Here is my current converter:
```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const e = exposes.presets;

module.exports = [
	{
        zigbeeModel: ['TAFFETAS2 D1.00P1.02Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000', 'TAFFETAS2 D1.00P1.01Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000'],
        model: 'TAFFETAS2',
        vendor: 'Acova',
        description: 'Taffetas 2 heater (custom)',
        fromZigbee: [fz.thermostat, fz.hvac_user_interface, fz.occupancy],
        toZigbee: [
            tz.thermostat_local_temperature,
            tz.thermostat_system_mode,
            tz.thermostat_occupied_heating_setpoint,
            tz.thermostat_unoccupied_heating_setpoint,
            tz.thermostat_occupied_cooling_setpoint,
            tz.thermostat_running_state,
        ],
        exposes: [
            exposes.climate()
                .withSetpoint('occupied_heating_setpoint', 7, 28, 0.5)
                .withSetpoint('unoccupied_heating_setpoint', 7, 28, 0.5)
                .withLocalTemperature()
                .withSystemMode(['off', 'heat', 'auto'])
                .withRunningState(['idle', 'heat']),
            e.occupancy(),
        ],
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(1);
            const endpoint2 = device.getEndpoint(2);
            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 'hvacThermostat']);
            await reporting.bind(endpoint2, coordinatorEndpoint, ['msOccupancySensing']);
            await reporting.thermostatTemperature(endpoint);
            await reporting.thermostatRunningState(endpoint);
            await reporting.thermostatOccupiedHeatingSetpoint(endpoint);
            await reporting.thermostatUnoccupiedHeatingSetpoint(endpoint);
            await reporting.occupancy(endpoint2)
        },
    },
];
```
